osu.Game/Online/Multiplayer/MultiplayerClient.cs

```diff
@@ -701,28 +709,37 @@ private void updateLocalRoomSettings(MultiplayerRoomSettings settings)
             CurrentMatchPlayingItem.Value = APIRoom.Playlist.SingleOrDefault(p => p.ID == settings.PlaylistItemId);
         }
 
-        private async Task<PlaylistItem> createPlaylistItem(MultiplayerPlaylistItem item)
+        private async Task<PlaylistItem> createPlaylistItem(MultiplayerPlaylistItem item, bool populateBeatmapImmediately)
         {
-            var apiBeatmap = await GetAPIBeatmap(item.BeatmapID).ConfigureAwait(false);
-
             var ruleset = Rulesets.GetRuleset(item.RulesetID);
             var rulesetInstance = ruleset.CreateInstance();
 
             var playlistItem = new PlaylistItem
             {
                 ID = item.ID,
+                BeatmapID = item.BeatmapID,
                 OwnerID = item.OwnerID,
-                Beatmap = { Value = apiBeatmap },
                 Ruleset = { Value = ruleset },
                 Expired = item.Expired
             };
 
             playlistItem.RequiredMods.AddRange(item.RequiredMods.Select(m => m.ToMod(rulesetInstance)));
             playlistItem.AllowedMods.AddRange(item.AllowedMods.Select(m => m.ToMod(rulesetInstance)));
 
+            if (populateBeatmapImmediately)
+                await PopulateBeatmap(playlistItem).ConfigureAwait(false);
+
             return playlistItem;
         }
 
+        public async Task PopulateBeatmap(PlaylistItem item)
+        {
+            if (item.Beatmap.Value != null)
+                return;
+
+            item.Beatmap.Value = await GetAPIBeatmap(item.BeatmapID).ConfigureAwait(false);
```

Comment:
Is this also a potential issue in that case? Having the `Beatmap` bindable changed from an arbitrary thread? Or only `APIRoom` that we care about?

Reply:
Oh, absolutely.